### PR TITLE
drop unused variable in API

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

dropping some unused arguments from APIP which does not affect user

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/rpc_sequential.py [hunk 1]
```diff
@@ -266,17 +266,17 @@ def configure_ddp(self):
...
+        current_layers = model.sequential_module
         model.sequential_module.foreach_worker(
             save_layers_on_all_rank_zero_workers, {"gpus_per_model": self.gpus_per_model}, include_self=True
         )
-        pl_module.sequential_module = load_sequential_from_saved_layers(self.gpus_per_model)
-        save_model_fn(last_filepath, trainer, pl_module)
-        pl_module.sequential_module = current_layers
+        model.sequential_module = load_sequential_from_saved_layers(self.gpus_per_model)
+        save_model_fn(last_filepath, trainer)
+        model.sequential_module = current_layers
```
Comment (Borda) [inline pytorch_lightning/plugins/training_type/rpc_sequential.py:279]: not sure how much this makes sense since the `model` is not used...

**Review by s-rog — APPROVED** [review]

**Comment by carmocca** [comment]
The changes here to `rpc` and `ModelCheckpoint` are also in https://github.com/PyTorchLightning/pytorch-lightning/pull/6136. I'd rather get that merged first since it's a P0

File: pytorch_lightning/plugins/training_type/rpc_sequential.py (hunk 1 above)
Comment (tchaton) [reply pytorch_lightning/plugins/training_type/rpc_sequential.py:279]: Yes, it is used in `save_model_fn` as trainer will access `lightning_module`. So we re-create full model only temporary. But your changes should be fine.

**Review by akihironitta — COMMENTED** [review]
Could we remove `pl_module` from the following private methods, too?:
https://github.com/PyTorchLightning/pytorch-lightning/blob/424b71e145125a0c68ae1c56b3bd4c20d71cee85/pytorch_lightning/callbacks/model_checkpoint.py#L541
https://github.com/PyTorchLightning/pytorch-lightning/blob/424b71e145125a0c68ae1c56b3bd4c20d71cee85/pytorch_lightning/callbacks/model_checkpoint.py#L520

Also, there are still outdated comments `# Todo: required argument pl_module is not used` in the file `model_checkpoint.py`.

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**